tests/test_crud_immunisation_api.py

```diff
@@ -302,51 +317,75 @@ def test_get_s_flag_patient(nhsd_apim_proxy_url, nhsd_apim_auth_headers, nhs_num
         assert retrieved_get_imms_result.status_code == 200
     retrieved_get_imms = retrieved_get_imms_result.json()
 
-    sample_disease_code = 840539006
-    retrieved_search_imms_result = imms_api.search_immunizations(nhs_number, sample_disease_code)
+    sample_disease_code = "COVID-19"
+    retrieved_search_imms_result = imms_api.search_immunizations(
+        nhs_number, sample_disease_code
+    )
     if retrieved_search_imms_result.status_code != 200:
         pprint.pprint(retrieved_search_imms_result.text)
         assert retrieved_search_imms_result.status_code == 200
-    retrieved_search_imms = next(imms for imms in retrieved_search_imms_result.json()["entry"]
-                                 if imms["resource"]["id"] == created_imms["id"])
-    # Fetching Immunization resource form Bundle
-    retrieved_search_imms = retrieved_search_imms["resource"]
+    retrieved_search_imms = next(
+        imms
+        for imms in retrieved_search_imms_result.json()["entry"]
+        if imms["id"] == created_imms["id"]
+    )
+
     all_retrieved_imms = [retrieved_get_imms, retrieved_search_imms]
+
     imms_api.delete_immunization(created_imms["id"])
 
     # Assert
     def get_questionnaire_items(imms):
-        questionnaire = next(contained for contained in imms["contained"]
-                             if contained["questionnaire"] == "Questionnaire/1")
+        questionnaire = next(
+            contained
+            for contained in imms["contained"]
+            if contained["resourceType"] == "QuestionnaireResponse"
+        )
         return questionnaire["item"]
 
     def assert_is_not_filtered(imms):
         imms_items = get_questionnaire_items(imms)
 
-        for key in ["SiteName", "Consent"]:
+        for key in ["Consent"]:
             assert key in [item["linkId"] for item in imms_items]
 
-        assert "N2N9I" != next(item for item in imms_items
```

Comment:
Why has this been changed to a single? This test has no knowledge of other tests or validation checking for multiple Organizations. It needs to assert that any Organizations that may be here are not showing filtered fields.

Reply:
fixed